pkg/scheduler/plugins/elasticquota/quota_overuse_revoke.go

```diff
@@ -0,0 +1,272 @@
+		if shouldBreak, _ := quotav1.LessThanOrEqual(used, runtime); shouldBreak {
+			break
+		}
+
+		pod := iter.Value().(*v1.Pod)
+		podReq, _ := resource.PodRequestsAndLimits(pod)
+		used = quotav1.Subtract(used, podReq)
+		tryAssignBackPodCache = append(tryAssignBackPodCache, pod)
+	}
+
+	// means should evict all
+	if lessThanOrEqual, _ := quotav1.LessThanOrEqual(used, runtime); !lessThanOrEqual {
+		for _, pod := range tryAssignBackPodCache {
+			klog.Infof("pod should be revoked by QuotaOverUsedMonitor, pod:%v, quotaName:%v"+
+				"used:%v, runtime:%v", pod.Name, quotaName, oriUsed, runtime)
+		}
+		return tryAssignBackPodCache
+	}
+
+	//try assign back from high->low
+	realRevokePodCache := make([]*v1.Pod, 0)
+	for index := len(tryAssignBackPodCache) - 1; index >= 0; index-- {
+		func() {
+			pod := tryAssignBackPodCache[index]
+			podRequest, _ := resource.PodRequestsAndLimits(pod)
+
+			used = quotav1.Add(used, podRequest)
+			if canAssignBack, _ := quotav1.LessThanOrEqual(used, runtime); !canAssignBack {
+				used = quotav1.Subtract(used, podRequest)
+				realRevokePodCache = append(realRevokePodCache, pod)
+			}
+		}()
+	}
+	for _, pod := range realRevokePodCache {
+		klog.Infof("pod should be evict by QuotaOverUseGroupMonitor, pod:%v, quotaName:%v,"+
+			"used:%v, runtime:%v", pod.Name, quotaName, oriUsed, runtime)
+	}
+	return realRevokePodCache
+}
+
+func PodPriorityComparator(a, b interface{}) int {
+	lhs := a.(*v1.Pod)
+	rhs := b.(*v1.Pod)
+
+	if corev1.PodPriority(lhs) > corev1.PodPriority(rhs) {
+		return 1
+	} else if corev1.PodPriority(lhs) < corev1.PodPriority(rhs) {
+		return -1
+	}
+
+	return 0
+}
+
+type QuotaOverUsedRevokeController struct {
+	clientSet                         clientset.Interface
+	groupQuotaManger                  *core.GroupQuotaManager
+	group2MonitorMapLock              sync.RWMutex
+	group2MonitorMap                  map[string]*QuotaOverUsedGroupMonitor
+	continueOverUsedCountTriggerEvict int64
+	monitorAllQuotas                  bool
+}
+
+func NewQuotaOverUsedRevokeController(client clientset.Interface, continueOverUsedCountTriggerEvict int64,
+	groupQuotaManager *core.GroupQuotaManager, monitorAllQuotas bool) *QuotaOverUsedRevokeController {
+	controller := &QuotaOverUsedRevokeController{
+		clientSet:                         client,
+		groupQuotaManger:                  groupQuotaManager,
+		continueOverUsedCountTriggerEvict: continueOverUsedCountTriggerEvict,
+		group2MonitorMap:                  make(map[string]*QuotaOverUsedGroupMonitor),
+		monitorAllQuotas:                  monitorAllQuotas,
+	}
+	return controller
+}
+
+func (controller *QuotaOverUsedRevokeController) Start() {
+	go wait.Until(controller.RevokePodDueToQuotaOverUsed, RevokePodCycle, nil)
+}
+
+func (controller *QuotaOverUsedRevokeController) RevokePodDueToQuotaOverUsed() {
+	toRevokePods := controller.MonitorAll()
+	for _, pod := range toRevokePods {
+		if err := util.DeletePod(controller.clientSet, pod); err != nil {
```

Comment:
delete pod via the Eviction API  that can be protected by PDB or PUB if the Pod's workload are deploying.